src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -853,18 +858,22 @@ private static bool InsideNameOfExpression(ExpressionSyntax expr, SemanticModel
 
         private static bool PreferPredefinedTypeKeywordInDeclarations(NameSyntax name, OptionSet optionSet, SemanticModel semanticModel)
         {
-            return (name.Parent != null) && !(name.Parent is MemberAccessExpressionSyntax) &&
-                   !InsideCrefReference(name) && !InsideNameOfExpression(name, semanticModel) &&
-                   optionSet.GetOption(SimplificationOptions.PreferIntrinsicPredefinedTypeKeywordInDeclaration, LanguageNames.CSharp);
+            return !IsInMemberAccessContext(name) &&
+                   !InsideCrefReference(name) &&
+                   !InsideNameOfExpression(name, semanticModel) &&
+                   SimplificationHelpers.PreferPredefinedTypeKeywordInDeclarations(optionSet, semanticModel.Language);
         }
 
-        private static bool PreferPredefinedTypeKeywordInMemberAccess(ExpressionSyntax memberAccess, OptionSet optionSet, SemanticModel semanticModel)
+        private static bool PreferPredefinedTypeKeywordInMemberAccess(ExpressionSyntax expression, OptionSet optionSet, SemanticModel semanticModel)
         {
-            return (((memberAccess.Parent != null) && (memberAccess.Parent is MemberAccessExpressionSyntax)) || InsideCrefReference(memberAccess)) &&
-                   !InsideNameOfExpression(memberAccess, semanticModel) &&
-                   optionSet.GetOption(SimplificationOptions.PreferIntrinsicPredefinedTypeKeywordInMemberAccess, LanguageNames.CSharp);
+            return (IsInMemberAccessContext(expression) || InsideCrefReference(expression)) &&
+                   !InsideNameOfExpression(expression, semanticModel) &&
+                   SimplificationHelpers.PreferPredefinedTypeKeywordInMemberAccess(optionSet, semanticModel.Language);
         }
 
+        public static bool IsInMemberAccessContext(this ExpressionSyntax expression) =>
+            (expression?.Parent is MemberAccessExpressionSyntax);
```

Comment:
just refactored out existing code here. I'd like to keep the ?. just because it existed.